bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/DanfossAirUnitHandler.java

```diff
@@ -0,0 +1,165 @@
+
+    private final Logger logger = LoggerFactory.getLogger(DanfossAirUnitHandler.class);
+
+    private @Nullable DanfossAirUnitConfiguration config;
+
+    public DanfossAirUnitHandler(Thing thing) {
+        super(thing);
+    }
+
+    private @Nullable ValueCache valueCache;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private @Nullable DanfossAirUnit hrv;
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateAllChannels();
+        } else {
+            try {
+                DanfossAirUnit danfossAirUnit = hrv;
+                Channel channel = Channel.getByName(channelUID.getIdWithoutGroup());
+                if (channel.getWriteAccessor() != null) {
+                    updateState(channelUID, channel.getWriteAccessor().access(danfossAirUnit, command));
+                }
+            } catch (IllegalArgumentException e) {
+                logger.info("{}", e.getMessage());
+            } catch (IOException ioe) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, ioe.getMessage());
+            } catch (Exception e) {
+                logger.info("Error updating channel {}: {}", channelUID, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        config = getConfigAs(DanfossAirUnitConfiguration.class);
+        valueCache = new ValueCache(config.updateUnchangedValuesEveryMillis);
+        try {
+            hrv = new DanfossAirUnit(InetAddress.getByName(config.host), 30046);
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Unknown host: " + config.host);
+            return;
+        }
+
+        scheduler.execute(() -> {
+            try {
+                thing.setProperty(PROPERTY_UNIT_NAME, hrv.getUnitName());
+                thing.setProperty(PROPERTY_SERIAL, hrv.getUnitSerialNumber());
+                pollingJob = scheduler.scheduleWithFixedDelay(this::updateAllChannels, 5, config.refreshInterval,
+                        TimeUnit.SECONDS);
+                updateStatus(ThingStatus.ONLINE);
+
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+            }
+        });
+    }
+
+    private void updateAllChannels() {
+        if (hrv == null) {
+            return;
+        }
+        DanfossAirUnit danfossAirUnit = hrv;
+        logger.debug("Updating DanfossHRV data '{}'", getThing().getUID());
+
+        try {
+            for (Channel channel : Channel.values()) {
+                if (Thread.interrupted()) {
+                    logger.debug("Polling thread interrupted...");
+                    return;
+                }
+                try {
```

Comment:
If you really need this nested try-catch I would suggest to split of the part of the code that contains the nested try-catch to a separate method. This makes your code more readable and it might also make it more clear which part of the exceptions should be handled inside the method and which in the `updateAllChannels` method